### PR TITLE
Fixes for unpack tarballs

### PR DESCRIPTION
**Pull request by portante**
First, we remove `pbench-move-unpacked`. It is no longer used in any of our deployments, and it is a hassle to keep around the code. Removing it now is less painful than trying to remove it later when other code changes may be necessary.

Second, we now fix the order in which operations performed. The indexing and the copying of sosreports now follows unpacking of tar balls. This removes a race condition where `pbench-copy-sosreports` may miss seeing sosreports because the tar ball is not unpacked. It also gives us the opportunity to leverage the already unpacked tar balls for the indexing step (planned for a future commit).

This commit depends on PRs #1294 and #1295.

### CODE REVIEW

File: server/bin/state/config-satellite/pbench-server.cfg [hunk 1]
```diff
@@ -11,6 +11,7 @@ pbench-top-dir = %(unittest-dir)s/pbench-satellite
 pbench-local-dir = %(unittest-dir)s/pbench-satellite-local
 # A satellite just unpacks and syncs
 dispatch-states = TO-UNPACK, TO-SYNC
+unpacked-states = UNPACKED
```
Comment (ndokos) [inline server/bin/state/config-satellite/pbench-server.cfg]: UNPACKED is the standard link desitnation for pbench-unpack-tarballs, I think `unpacked-states` should be all the others that depend on the unpacked tarball being there: TO-INDEX etc.

Comment (portante) [reply server/bin/state/config-satellite/pbench-server.cfg]: Good catch.

File: server/lib/config/pbench-server-default.cfg [hunk 2]
```diff
@@ -83,10 +83,10 @@ use-mail-for-status = no
 # When a tarball comes in to this server, what processing should it
 # undergo?  Production-like servers do at least the first two, and
 # possibly more, of the following dispatch states:
-#dispatch-states = TO-UNPACK, TO-INDEX, TO-COPY-SOS, TO-BACKUP
+dispatch-states = TO-UNPACK, TO-INDEX, TO-COPY-SOS, TO-BACKUP
 
 # Satellite servers typically do at least this dispatch state:
```
[the remark this answers is not in the document]
Comment (portante) [reply server/lib/config/pbench-server-default.cfg]: Fixed.

File: server/bin/state/config-satellite/pbench-server.cfg (hunk 1 above)
Comment (portante) [reply server/bin/state/config-satellite/pbench-server.cfg]: Fixed.

**Pull request merged**